encled

```diff
@@ -153,6 +160,38 @@ def list_all():
             devlist.append((e, s, path, status))
     return devlist
 
+# Recursively find SCSI devices in block device stacking hierarchies
+# by referencing /sys/block/<dev>/slaves/*
+def find_parent_devs(dev_names):
```

Comment:
I like this idea, but I don't like updates to the dev_names in the code. It's really hard to reason about it (it passed as argument into recursive function and is updating it's arguments). 

Can you change it in a way, which does not update a function argument? A new variable like 'combined = dev_name + new_names` would do it.

Second, there is a global depth, but I don't see use of it. (debug leftovers?).